### ISSUE TITLE
Correct # of edges?

### ISSUE THREAD

**Issue by tsafavi** [MPeti01/wikipedia-gnns#3]
Hi, thanks for contributing this dataset! 

I was wondering what is the correct number of edges for the graph. The paper reports 216123, but when I process the dataset myself I get 297110 edges. Also, when I load the `WikiCS` dataset with pytorch-geometric, the edge_index is a tensor of shape [2, 297110], and `num_edges` also returns 297110. Could you clarify?

Thanks again,
Tara

**Comment by pmernyei** [comment]
Hi, sorry for the slow response - I don't have an immediate answer but I plan to take a look later this week, that discrepancy does seem strange. Note that the edge_index tensor contains two entries for a single edge in two directions for undirected graph, but that just gives us a discrepancy in the other direction. One idea that seems possible: maybe I incorrectly counted the edges before turning it into an undirected graph, and mistakenly counted two-cycles in the original link graph as two edges even though it gets turned to a single edge.

**Comment by pmernyei** [comment]
Hi! My speculation above was vaguely along the right track, it's about directionality. Looking at the dataset as a directed graph, there are 297110 edges, but if you look at each edge as undirected (hence (i,j) = (j,i)) you are left with 216123 edges (including 520 loops).

Since I originally intended the dataset to be used as an undirected graph, it's unfortunate that I stored the directed edges in `data.json` and 1) I neglected to process to undirected edges in one of the two data loader implementations in this repo, 2) the data loader in pytorch geometric also loads them as directed edges. I'm not sure what makes the most sense at this point. I'll have a look at how people are using the dataset - if they also seem to be intending to deal with undirected graphs, I'll probably update this repo and get pytorch geometric updated as well. In the meantime, I suggest you consider your use-case and process the data yourself accordingly.
